packages/gitmoji-changelog-markdown/src/index.js

```diff
@@ -31,51 +32,51 @@ function toMarkdown({ meta, changes }) {
 }
 
 function markdownFromScratch({ meta, changes }, options) {
-  const output = toMarkdown({ meta, changes })
-  return Promise.resolve(fs.writeFileSync(options.output, output))
+  return promisify(fs.writeFile)(options.output, `# Changelog\n\n${toMarkdown({ meta, changes })}`)
 }
 
 function markdownIncremental({ meta, changes }, options) {
   const { lastVersion } = meta
-
-  const currentFile = options.output
-  const tempFile = `${currentFile}.tmp`
-
-  if (!fs.existsSync(currentFile)) {
-    throw new Error(`${currentFile} doesn't exists, please execute "gitmoji-changelog init" to build it from scratch.`)
-  }
-
-  // write file for next version
-  const writer = fs.createWriteStream(tempFile, { encoding: 'utf-8' })
-  writer.write(toMarkdown({ meta, changes }))
-
-  // read original file until last tags and add it to the end
-  let lastVersionFound = false
-  return new Promise(resolve => {
-    const stream = fs.createReadStream(currentFile)
-      .pipe(es.split())
-      .pipe(es.mapSync((line) => {
-        stream.pause()
-
-        if (line.startsWith(`<a name="${lastVersion}"></a>`)) {
-          lastVersionFound = true
-        }
-        if (lastVersionFound) {
-          writer.write(`${line}\n`)
-        }
-
-        stream.resume()
+  const { output, release } = options
+
+  const tempFile = `${output}.tmp`
+
+  return new Promise((resolve, reject) => {
+    const readStream = fs.createReadStream(output, { encoding: 'utf-8' })
+    const writeStream = fs.createWriteStream(tempFile, { encoding: 'utf-8' })
+
+    // start from currentFile
+    readStream
+      .pipe(new Transform({
+        transform(chunk, encoding, callback) {
+          const string = chunk.toString()
+
+          // if the release (the next) is already in file, we do nothing
+          if (string.includes(`<a name="${release}"></a>`)) {
```

Comment:
It would be better to filter lines out, to rewrite them (in case the next release has a new commit)

Reply:
done